replicate/common/generic-replicator.go

```diff
@@ -198,6 +200,50 @@ func (r *GenericReplicator) NamespaceAdded(ns *v1.Namespace) {
 	}
 }
 
+// NamespaceUpdated checks if namespace's labels changed and deletes any 'replicate-to-matching' resources
+// the namespace no longer qualifies for. Then it attempts to replicate resources into the updated ns based
+// on the updated set of labels
+func (r *GenericReplicator) NamespaceUpdated(nsOld *v1.Namespace, nsNew *v1.Namespace) {
+	logger := log.WithField("kind", r.Kind).WithField("target", nsNew.Name)
+	// check if labels changed
+	labelsEqual := reflect.DeepEqual(nsNew.Labels, nsOld.Labels)
+	if  labelsEqual{
+		logger.Debug("labels didn't change")
```

Comment:
This could be shortened to an early exit, to reduce the overall nesting depth of this function:

```go
if reflect.DeepEqual(nsNew.Labels, nsOld.Labels) {
    logger.Debug("labels didn't change")
    return
}